providers/implementations/keymgmt/ml_kem_kmgmt.c

```diff
@@ -125,11 +127,21 @@ static int ml_kem_pairwise_test(const ML_KEM_KEY *key)
 }
 #endif  /* FIPS_MODULE */
 
-static void *ml_kem_new(OSSL_LIB_CTX *libctx, char *propq, int evp_type)
+static void *ml_kem_new(void *vctx, int evp_type)
 {
+    const OSSL_PROVIDER *prov;
+    OSSL_LIB_CTX *libctx = NULL;
+    int retain_seed = 1;
+
     if (!ossl_prov_is_running())
         return NULL;
-    return ossl_ml_kem_key_new(libctx, propq, evp_type);
+    if (vctx != NULL) {
+        libctx = PROV_LIBCTX_OF(vctx);
+        prov = (OSSL_PROVIDER *)ossl_prov_ctx_get0_handle(vctx);
+        retain_seed =
+            OSSL_PROVIDER_conf_get_bool(prov, "pkey_retain_seed", 1);
+    }
+    return ossl_ml_kem_key_new(libctx, NULL, retain_seed, evp_type);
```

Comment:
I am assuming there is always the opportunity to set the propq after the ml_kem_new() is called?

Reply:
No, there isn't.

These affect hash function selection and RNG choice, but b/c export does not set the `OSSL_PKEY_PARAM_PROPERTIES` value, and in any case, that'd be from the source provider and may not be right for the target, ...

There just isn't another source for the props when not initially set as part of keygen. We do use `OSSL_PKEY_PARAM_PROPERTIES` in `gen_init`, but not on import or load which don't provide a value to use.